amplify/src/main/java/com/github/stkent/amplify/tracking/AmplifyStateTracker.java

```diff
@@ -178,9 +148,10 @@ public void promptIfReady(@NonNull final AmplifyView amplifyView) {
 
     public boolean shouldAskForRating() {
         return allEnvironmentRequirementsMet()
-                && allTotalEventCountPredicatesAllowFeedbackPrompt()
-                && allLastEventTimePredicatesAllowFeedbackPrompt()
-                && allLastEventVersionPredicatesAllowFeedbackPrompt();
+                && totalEventCountPredicates.allowFeedbackPrompt()
```

Comment:
Sounds good to me. I wouldn't dare try to come up with the names myself so what are you thinking?